### ISSUE TITLE
(🐞) errors from imported modules aren't reported when project is isntalled

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#18106]
## given
```
-src
 |- __init__.py
 |- test.py
 |- test2.py
```
```py
# src/test.py
from src import test2

reveal_type(test2.a)
```
```py
# src/test2.py
1 + ""
a = 1
```
## then
```
> mypy -m src.test
src/test.py:4: note: Revealed type is "builtins.int"
Success: no issues found in 1 source file
> mypy src/test
src/test.py:4: note: Revealed type is "builtins.int"
Success: no issues found in 1 source file
```

# expected 

[reference](https://mypy.readthedocs.io/en/stable/running_mypy.html#following-imports) 

> Mypy is designed to [doggedly follow all imports](https://mypy.readthedocs.io/en/stable/running_mypy.html#finding-imports), even if the imported module is not a file you explicitly wanted mypy to check.

> [`--follow-imports`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-follow-imports)`=normal` follows all imports normally and type checks all top level code (as well as the bodies of all functions and methods with at least one type annotation in the signature).

```
> mypy src/test.py
src/test2.py:1: error: Unsupported operand types for + ("int" and "str")  [operator]
src/test.py:4: note: Revealed type is "builtins.int"
Found 1 error in 1 file (checked 3 source files)
```

> type checks all code

where are the errors from this typechecking?

# i found them

```
> mypy --no-silence-site-packages -m src.test
src/test2.py:1: error: Unsupported operand types for + ("int" and "str")  [operator]
src/test.py:4: note: Revealed type is "builtins.int"
```

uhhhh, what? 

> Enables reporting error messages generated within installed packages (see [PEP 561](https://peps.python.org/pep-0561/) for more details on distributing type information). Those error messages are suppressed by default, since you are usually not able to control errors in 3rd party code.

```
> mypy -m src.test
src/test.py:4: note: Revealed type is "builtins.int"
Success: no issues found in 1 source file
> uv pip uninstall -e .
...
> mypy -m src.test
```
